src/compiler/parser.ts

```diff
@@ -1001,10 +1008,12 @@ module ts {
             switch (parsingContext) {
                 case ParsingContext.SourceElements:
                 case ParsingContext.ModuleElements:
-                    return isSourceElement(inErrorRecovery);
+                    // During error recovery we don't treat empty statements as statements

```

Comment:
I would prefer hte original comment remains.  This comment is just the english translation of the code below it.  It doesn't indicate the part about why this is important.  Someone reading this in the future could see this and ask "Buy why aren't empty statements treated as statements in error recovery?  Is that correct?  Is that desirable?  Is that the cause of this bug that i'm investigating?"

Code can be 100% correct, and yet also not be great at being self-documenting.  In cases like that, documentation goes a long way toward maintainability and clarity.